src/olympia/bandwagon/tests/test_admin.py

```diff
@@ -78,6 +133,74 @@ def test_can_not_edit_without_collections_edit_permission(self):
         assert collection.slug == 'floob'
 
+    def test_can_do_limited_edition_with_admin_curation_permission(self):
+        collection = Collection.objects.create(slug='floob')
+        self.detail_url = reverse(
+            'admin:bandwagon_collection_change', args=(collection.pk,)
+        )
+        user = user_factory()
+        self.grant_permission(user, 'Admin:Tools')
+        self.grant_permission(user, 'Admin:Curation')
+        self.client.login(email=user.email)
+        response = self.client.get(self.detail_url, follow=True)
+        assert response.status_code == 403
+        assert collection.slug not in response.content.decode('utf-8')
+
+        post_data = {
+            # Django wants the whole form to be submitted, unfortunately.
+            'application': amo.FIREFOX.id,
+            'type': collection.type,
+            'default_locale': collection.default_locale,
+            'author': user.pk,
+        }
+        post_data['slug'] = 'bar'
+        response = self.client.post(self.detail_url, post_data, follow=True)
+        assert response.status_code == 403
+        collection.reload()
+        assert collection.slug == 'floob'
+
+        # Now, if it's a mozilla collection, you can edit it.
+        mozilla = user_factory(username='mozilla')
+        collection.update(author=mozilla)
+        response = self.client.get(self.detail_url, follow=True)
+        assert response.status_code == 200
+        assert collection.slug in response.content.decode('utf-8')
+
+        post_data = {
+            # Django wants the whole form to be submitted, unfortunately.
+            'application': amo.FIREFOX.id,
+            'type': collection.type,
+            'default_locale': collection.default_locale,
+            'author': mozilla.pk,
+        }
+        post_data['slug'] = 'bar'
+        response = self.client.post(self.detail_url, post_data, follow=True)
+        assert response.status_code == 200
+        collection.reload()
+        assert collection.slug == 'bar'
+        assert collection.author.pk == mozilla.pk
+
+        # You can also edit it if it's your own (allowing you, amongst other
+        # things, to transfer it to mozilla)
+        collection.update(author=user)
+        response = self.client.get(self.detail_url, follow=True)
+        assert response.status_code == 200
+        assert collection.slug in response.content.decode('utf-8')
+
+        post_data = {
+            # Django wants the whole form to be submitted, unfortunately.
+            'application': amo.FIREFOX.id,
+            'type': collection.type,
+            'default_locale': collection.default_locale,
+            'author': mozilla.pk,
+        }
+        post_data['slug'] = 'fox'
+        response = self.client.post(self.detail_url, post_data, follow=True)
+        assert response.status_code == 200
+        collection.reload()
+        assert collection.slug == 'fox'
+        assert collection.author.pk == mozilla.pk
+
```

Comment:
Is there a test that checks the curator can transfer ownership of the collection?

Reply:
In the middle of this test the collection author is changed, yes.